docs/adding-new-resource.md

```diff
@@ -0,0 +1,705 @@
+name>/<node name>`. We will need to tell Upjet to use `<node name>` as external
+name and take the rest from the parameters.
+
+Use `config.TemplatedStringAsIdentifier("<name argument>", "<go template>")` in
+such cases. The following is the list of available parameters for you to use in
+your go template:
+
+```
+parameters: A tree of parameters that you'd normally see in a Terraform HCL
+            file. You can use TF registry documentation of given resource to
+            see what's available.
+
+terraformProviderConfig: The Terraform configuration object of the provider. You can
+                take a look at the TF registry provider configuration object
+                to see what's available. Not to be confused with ProviderConfig
+                custom resource of the Crossplane provider.
+
+externalName: The value of external name annotation of the custom resource.
+              It is required to use this as part of the template.
+```
+
+You can see example usages in the big three providers below.
+
+#### AWS
+
+For `aws_glue_user_defined_function`, we see that the `name` argument is used
+to name the resource and the import instructions read as the following:
+
+```
+Glue User Defined Functions can be imported using the
+`catalog_id:database_name:function_name`. If you have not set a Catalog ID
+specify the AWS Account ID that the database is in, e.g.,
+
+$ terraform import aws_glue_user_defined_function.func 123456789012:my_database:my_func
+```
+
+Our configuration would look like the following:
+
+```golang
+"aws_glue_user_defined_function":  config.TemplatedStringAsIdentifier("name", "{{ .parameters.catalog_id }}:{{ .parameters.database_name }}:{{ .externalName }}")
+```
+
+Another prevalent case in AWS is the usage of Amazon Resource Name (ARN) to
+identify a resource. We can use `config.TemplatedStringAsIdentifier` in many of
+those cases like the following:
+
+```
+"aws_glue_registry": config.TemplatedStringAsIdentifier("registry_name", "arn:aws:glue:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:registry/{{ .external_name }}"),
+```
+
+However, there are cases where the ARN includes random substring and that would
+fall under Case 4. The following is such an example:
+
+```
+// arn:aws:acm-pca:eu-central-1:609897127049:certificate-authority/ba0c7989-9641-4f36-a033-dee60121d595
+	"aws_acmpca_certificate_authority_certificate": config.IdentifierFromProvider,
+```
+
+#### Azure
+
+Most Azure resources fall under this case since they use fully qualified
+identifier as Terraform ID.
+
+For `azurerm_mariadb_firewall_rule`, we see that the `name` argument is used to
+name the resource and the import instructions read as the following:
+
+```
+MariaDB Firewall rules can be imported using the resource ID, e.g.
+
+terraform import azurerm_mariadb_firewall_rule.rule1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1/firewallRules/rule1
+```
```

Comment:
```suggestion
> MariaDB Firewall rules can be imported using the resource ID, e.g.
>
> `terraform import azurerm_mariadb_firewall_rule.rule1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1/firewallRules/rule1`
```

Suggestion: A block quote seems more appropriate here and has the additional benefit of formatting nested code properly.